hoodie-client/src/main/java/com/uber/hoodie/io/compact/strategy/DayBasedCompactionStrategy.java

```diff
@@ -57,8 +57,10 @@
   @Override
   public List<CompactionOperation> orderAndFilter(HoodieWriteConfig writeConfig,
       List<CompactionOperation> operations) {
-    // Iterate through the operations and accept operations as long as we are within the IO limit
+    // Iterate through the operations and accept operations as long as we are within the configured target partitions
+    // limit
     return super.orderAndFilter(writeConfig,
-        operations.stream().sorted(comparator).collect(Collectors.toList()));
+        operations.stream().sorted(comparator).collect(Collectors.toList()))
+        .subList(0, writeConfig.getTargetPartitionsPerCompactionInMB());
```

Comment:
This does not look correct to me. CompactionOperation is at the file-slice level (not partition). As of this point the CompactionOperations are reverse-sorted but there could be multiple operations within a partition.  The filter logic cannot be just sublist right ? 
 It has to inspect the partitionPath of CompactionOperation when deciding to allow or not.  Can you cross-check ?

Reply:
file-slice => fileId in the above comment